src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/AndroidNetworkInterface.cs

```diff
@@ -0,0 +1,111 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
+using System.Text;
+using System.Net;
+
+namespace System.Net.NetworkInformation
+{
+    /// <summary>
+    /// Implements a NetworkInterface on Android.
+    /// </summary>
+    internal sealed class AndroidNetworkInterface : UnixNetworkInterface
+    {
+        private OperationalStatus _operationalStatus;
+        private bool _supportsMulticast;
+        private long _speed;
+        internal int _mtu;
+        private NetworkInterfaceType _interfaceType = NetworkInterfaceType.Unknown;
+        private readonly AndroidIPInterfaceProperties _ipProperties;
+
+        internal AndroidNetworkInterface(string name, int index) : base(name)
+        {
+            _index = index;
+            _ipProperties = new AndroidIPInterfaceProperties(this);
+        }
+
+        public static unsafe NetworkInterface[] GetAndroidNetworkInterfaces()
+        {
+            int interfaceCount=0;
+            int addressCount=0;
+            Interop.Sys.NetworkInterfaceInfo * nii = null;
+            Interop.Sys.IpAddressInfo * ai = null;
+            IntPtr globalMemory = (IntPtr)null;
+
+            if (Interop.Sys.GetNetworkInterfaces(&interfaceCount, &nii, &addressCount, &ai) != 0)
+            {
+                string message = Interop.Sys.GetLastErrorInfo().GetErrorMessage();
+                throw new NetworkInformationException(message);
+            }
+
+            globalMemory = (IntPtr)nii;
+            try
+            {
+                NetworkInterface[] interfaces = new NetworkInterface[interfaceCount];
+                Dictionary<int, AndroidNetworkInterface> interfacesByIndex = new Dictionary<int, AndroidNetworkInterface>(interfaceCount);
```

Comment:
Do you expect android devices to have that many interfaces that you need both an array and dictionary?

Reply:
This code was taken from the `LinuxNetworkInterface` implementation and I made only changes that were necessary, so I kept the Dictionary. I'll simplify it.